Code/Tools/AssetProcessor/native/tests/assetmanager/AssetProcessorManagerTest.cpp

```diff
@@ -4226,12 +4226,20 @@ TEST_F(LockedFileTest, DeleteFile_LockedProduct_DeletesWhenReleased)
     if (!AZ::IO::SystemFile::Delete(productPath.toUtf8().constData()))
     {
         AZStd::thread workerThread;
+        bool threadStarted = false;
 
-        m_callback = [&product, &workerThread]() {
-            workerThread = AZStd::thread([&product]() {
-                AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(60));
-                product.close();
-            });
+        m_callback = [&product, &workerThread, &threadStarted]() {
```

Comment:
the callback is passing the address of a variable in the heap, but the callback is a member variable
if the callback is called out of the scope of this function, it will override the heap
make the `threadStarted` member variable a member variable as well (preferred), or clear `m_callback` at the end of this function (I suggest doing this as well for all the tests, moreover, I would remove `m_callback` from being a member variable, that seems dangerous)